test/features/checkin-event.js

```diff
@@ -0,0 +1,286 @@
+'use strict';
+
+Feature('Check In Event', () => {
+
+  Scenario('Non-Existent DetaineeEvent should be Created', () => {
+
+    var date = new Date();
+    var dateString = date.toISOString();
+    var payload = {
+      person_id: 123,
+      cid_id: 100,
+      operation: 'check in',
+      timestamp: dateString,
+      gender: 'm',
+      nationality: 'gbr',
+      centre: 'abc'
+    };
+
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    Given('a check in event with timestamp "' + dateString + '" has not already occurred', (done) =>
+
+      global.initializeBarrelsFixtures(function () {
```

Comment:
why does this need to be inside this global function definition?
could this function not just be:

```
Given('a check in event with timestamp "' + dateString + '" has not already occurred', () =>
  DetaineeEvent.find({
    where: {
      timestamp: dateString,
      operation: 'check in'
    }
  }).then((models) => 
    expect(models.length).to.equal(0)
  )
);
```